src/nwb_conversion_tools/tools/roiextractors/roiextractors.py

```diff
@@ -95,52 +85,50 @@ def add_two_photon_series(imaging, nwbfile, metadata, buffer_size=10, use_times=
-        device = nwbfile.devices[list(nwbfile.devices.keys())[0]]
-        metadata["Ophys"]["ImagingPlane"][0]["optical_channel"] = [
-            OpticalChannel(**i) for i in metadata["Ophys"]["ImagingPlane"][0]["optical_channel"]
-        ]
-        metadata["Ophys"]["ImagingPlane"][0] = safe_update(metadata["Ophys"]["ImagingPlane"][0], {"device": device})
+    metadata = dict_deep_update(get_nwb_imaging_metadata(imaging), metadata)
+
+    # Tests if TwoPhotonSeries already exists in acquisition
+    two_photon_series_metadata = metadata["Ophys"]["TwoPhotonSeries"][0]
+    two_photon_series_name = two_photon_series_metadata["name"]
+    acquisition_modules = [module for module in nwbfile.acquisition]
 
-        imaging_plane = nwbfile.create_imaging_plane(**metadata["Ophys"]["ImagingPlane"][0])
+    # Only add if TwoPhotonSeries is not present before
+    if two_photon_series_name not in acquisition_modules:
 
+        # Add the data
         def data_generator(imaging):
             for i in range(imaging.get_num_frames()):
-                yield imaging.get_frames(frame_idxs=[i]).T
+                yield imaging.get_frames(frame_idxs=[i]).squeeze().T
 
         data = H5DataIO(
             DataChunkIterator(data_generator(imaging), buffer_size=buffer_size),
             compression=True,
         )
+        two_p_series_kwargs = two_photon_series_metadata
+        two_p_series_kwargs.update(data=data)
 
-        # using internal data. this data will be stored inside the NWB file
-        two_p_series_kwargs = dict_deep_update(
-            metadata["Ophys"]["TwoPhotonSeries"][0],
-            dict(data=data, imaging_plane=imaging_plane),
-        )
+        # Add the image plane
+        image_plane_metadata = metadata["Ophys"]["ImagingPlane"][0]
+        image_plane_metadata["optical_channel"] = [
+            OpticalChannel(**metadata) for metadata in image_plane_metadata["optical_channel"]
+        ]
 
-        if not use_times:
-            two_p_series_kwargs.update(
-                starting_time=imaging.frame_to_time(0),
-                rate=float(imaging.get_sampling_frequency()),
-            )
-        else:
-            two_p_series_kwargs.update(
-                timestamps=H5DataIO(
-                    imaging.frame_to_time(np.arange(imaging.get_num_frames())),
-                    compression="gzip",
-                )
-            )
+        device = nwbfile.devices[list(nwbfile.devices.keys())[0]]
+        image_plane_metadata["device"] = device
+        imaging_plane = nwbfile.create_imaging_plane(**image_plane_metadata)
+        two_p_series_kwargs.update(imaging_plane=imaging_plane)
+
+        # Add timestamps or rate
+        timestamps = imaging.frame_to_time(np.arange(imaging.get_num_frames()))
+        if use_times:
+            two_p_series_kwargs.update(timestamps=H5DataIO(timestamps, compression="gzip"))
             if "rate" in two_p_series_kwargs:
                 del two_p_series_kwargs["rate"]
```

Comment:
@h-mayorquin 
How about we put a warning here that using timestamps and rate is not supported and that rate will be removed?